core/src/test/java/org/infinispan/statetransfer/StateConsumerTest.java

```diff
@@ -264,14 +266,25 @@ public void test1() throws Exception {
       assertTrue(stateConsumer.hasActiveTransfers());
       assertEquals(flatRequestedSegments, newSegments);
 
-      // apply state
-      ArrayList<StateChunk> stateChunks = new ArrayList<>();
-      for (Integer segment : newSegments) {
-         stateChunks.add(new StateChunk(segment, Collections.emptyList(), true));
+      // We count how many segments were requested and then start to apply the state individually, to assert that the
+      // number of in-flight requests will decrease accordingly. During real usage, the state chunk collections can
+      // have more than a single segment.
+      long inflightCounter = requestedSegments.values().stream().mapToLong(Collection::size).sum();
+      assertEquals(stateConsumer.inflightRequestCount(), inflightCounter);
+      for (Map.Entry<Address, Set<Integer>> entry : requestedSegments.entrySet()) {
+         for (Integer segment : entry.getValue()) {
+            Collection<StateChunk> chunks = Collections.singletonList(new StateChunk(segment, Collections.emptyList(), true));
```

Comment:
While this test is great, I am having flash backs as to that this won't actually happen in production currently. This is caused by [ISPN-10706](https://issues.redhat.com/browse/ISPN-10706) which should now be doable since [ISPN-13434](https://issues.redhat.com/browse/ISPN-13434) is now integrated.